src/azure-cli/azure/cli/command_modules/vm/tests/hybrid_2020_09_01/test_vm_defaults.py

```diff
@@ -172,21 +172,25 @@ def setUp(self):
         ns.location = None
         self.ns = ns
 
-    @mock.patch('azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet.List', _mock_network_client_with_existing_vnet)
     def test_no_matching_vnet(self):
-        self._set_ns('emptyrg', 'eastus')
-        _validate_vm_vmss_create_vnet(_get_test_cmd(), self.ns)
-        self.assertIsNone(self.ns.vnet_name)
-        self.assertIsNone(self.ns.subnet)
-        self.assertEqual(self.ns.vnet_type, 'new')
+        from importlib import import_module
+        vnet = import_module('azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet')
+        with mock.patch.object(vnet, 'List', _mock_network_client_with_existing_vnet):
+            self._set_ns('emptyrg', 'eastus')
+            _validate_vm_vmss_create_vnet(_get_test_cmd(), self.ns)
+            self.assertIsNone(self.ns.vnet_name)
+            self.assertIsNone(self.ns.subnet)
+            self.assertEqual(self.ns.vnet_type, 'new')
 
-    @mock.patch('azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet.List', _mock_network_client_with_existing_vnet_location)
     def test_matching_vnet_specified_location(self):
-        self._set_ns('rg1', 'eastus')
-        _validate_vm_vmss_create_vnet(_get_test_cmd(), self.ns)
-        self.assertEqual(self.ns.vnet_name, 'vnet1')
-        self.assertEqual(self.ns.subnet, 'vnet1subnet')
-        self.assertEqual(self.ns.vnet_type, 'existing')
+        from importlib import import_module
+        vnet = import_module('azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet')
+        with mock.patch.object(vnet, 'List', _mock_network_client_with_existing_vnet_location):
+            self._set_ns('rg1', 'eastus')
+            _validate_vm_vmss_create_vnet(_get_test_cmd(), self.ns)
+            self.assertEqual(self.ns.vnet_name, 'vnet1')
+            self.assertEqual(self.ns.subnet, 'vnet1subnet')
+            self.assertEqual(self.ns.vnet_type, 'existing')
```

Comment:
Fix 
```
            _NAME_PATTERN = re.compile(f'^(?P<pkg>{dotted_words})'
                                       f'(?P<cln>:(?P<obj>{dotted_words})?)?$',
                                       re.UNICODE)

        m = _NAME_PATTERN.match(name)
        if not m:
>           raise ValueError(f'invalid format: {name!r}')
E           ValueError: invalid format: 'azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet'

C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1264.0_x64__qbz5n2kfra8p0\Lib\pkgutil.py:688: ValueError
----- generated xml file: C:\Users\hanglei\.azdev\env_config\Users\hanglei\Developer\azure-cli\venv\test_results.xml -----
================================================ short test summary info =================================================
FAILED src\azure-cli\azure\cli\command_modules\vm\tests\hybrid_2020_09_01\test_vm_defaults.py::TestVMCreateDefaultVnet::test_matching_vnet_specified_location - ValueError: invalid format: 'azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet'
```

This issue is also caused by the `mock` change in 3.11, which uses `pkgutil.resolve_name` instead of `importlib.import_module`. pkgutils requires that each part must be a valid python name, but `2020_09_01_hybrid` is not.

See https://github.com/python/cpython/issues/100950

Reply:
Why is `2020_09_01_hybrid` not valid? Can we fix the invalid name instead of changing the source code? Using `mock.patch.object` is only bypassing the issue, not solving it.

Reply:
In Python, variable name can't startwith digit.
Let's work with aaz to solve this.

Reply:
Solved by https://github.com/Azure/azure-cli/pull/27073

In summary, `import_module` allows module names starting with digits, but `pkgutil.resolve_name` and `import` don't.

```py
$ python -c 'import importlib; print(importlib.import_module("azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet"))'
<module 'azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet' from '/home/user2/azure-cli/src/azure-cli/azure/cli/command_modules/vm/aaz/2020_09_01_hybrid/network/vnet/__init__.py'>

$ python -c 'import pkgutil; print(pkgutil.resolve_name("azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet"))'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.10/pkgutil.py", line 687, in resolve_name
    raise ValueError(f'invalid format: {name!r}')
ValueError: invalid format: 'azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet'

$ python -c 'import azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet'
  File "<string>", line 1
    import azure.cli.command_modules.vm.aaz.2020_09_01_hybrid.network.vnet
                                                      ^
SyntaxError: invalid decimal literal
```